daffodil-core/src/main/scala/org/apache/daffodil/grammar/LocalElementGrammarMixin.scala

```diff
@@ -112,8 +112,11 @@ trait LocalElementGrammarMixin extends GrammarMixin { self: ElementBase =>
     // We don't know if this repeating thing is in trailing position, or in the middle
     // of a sequence. There is also ambiguity if the enclosing sequence and this sequence
     // have the same separator.
-    //      ~
-    //      RepAtMostTotalN(self, maxOccurs, separatedEmpty) // absorb extra separators, if found.
+    ~
+    (if (couldBeLastElementInModelGroup)
+      RepAtMostTotalN(self, maxOccurs, separatedEmpty) // absorb extra separators, if found.
+    else
+      EmptyGram)
     )
```

Comment:
Does this the issue mentioned in the comment? Seems like it does, and if so the comment should be removed?

Reply:
This should be addressed in the most recent commit